COLLABORATOR_GUIDE.md

```diff
@@ -73,6 +73,112 @@ All pull requests that modify executable code should be subjected to
+removal or redefinition of existing API arguments, changing return values,
+adding or removing errors, altering the timing, and changing the non-internal
+side effects of using a particular API.
+
+With a few notable exceptions outlined below, when backwards incompatible
+changes to an API are necessary, the existing API *must* be Deprecated *first*
+and the new API either introduced in parallel or added after the next major
+Node.js version following the Deprecation as a replacement for the Deprecated
+API. In other words, as a general rule, existing APIs *must not* change
+without a Deprecation.
+
+Exception to this rule is given in the following cases:
+
+* Adding or removing errors thrown or reported by an API;
+* Changing error messages;
+* Altering the timing and non-internal side effects of the API.
+
+Such changes *must* be handled as semver-major changes but MAY be landed
+without a [Deprecation cycle](#deprecation-cycle).
+
+From time-to-time, in particularly exceptional cases, the CTC may be asked to
+consider and approval additional exceptions to this rule.
+
+Purely additive changes (e.g. adding new events to EventEmitter 
+implementations, or adding new arguments to a method in a way that allows
+existing code to continue working without modification) are handled as
+semver-minor changes.
+
+Note that errors thrown by dependencies of Node.js (e.g. those originating from
+V8) are generally not under the control of Node.js and therefore are not
+directly subject to this policy. However, care should still be taken when
+landing updates to dependencies when it is known or expected that breaking
+changes to error handling may have been made. Additional CI testing may be
+required.
+
+### When breaking changes actually break things
+
+Breaking changes are difficult primarily because they change the fundamental
+assumptions a user of Node.js has when writing their code and can cause
+existing code to stop functioning as expected -- costing developers and users
+time and energy to fix.
+
+Because breaking (semver-major) changes are permitted to land in master at any
+time, it should be *understood and expected* that at least some subset of the
+user ecosystem *may* be adversely affected *in the short term* when attempting
+to build and use Node.js directly from master. This potential instability is
+precisely why Node.js offers distinct Stable and LTS release streams that carry
+varying explicit stability guarantees. Specifically, as a general rule,
+*breaking* changes should never land in a Stable or LTS branch unless such
+changes are considered to be absolutely necessary to address critical security
+issues.
+
+When any change is landed in master, and it is determined that the such
+changes *do*, in fact, break existing code, a decision may be made to revert
+those changes either temporarily or permanently. However, the decision to
+revert or not can often be based on many complex factors that are not easily
+codified.
```

Comment:
or, more often it may be tagged as semver-major after the fact if appropriate per the above guidlines